app/lib/apicast/curl_command_builder.rb

```diff
@@ -134,9 +134,8 @@ def builder_class
     end
 
     def proxy_from_config
-      proxy_configs = proxy.proxy_configs.by_environment(environment.to_s).current_versions.to_a
-      return if proxy_configs.empty?
-      ProxyFromConfig.new(proxy_configs.first.send(:parsed_content))
+      proxy_config = proxy.proxy_configs.by_environment(environment.to_s).current_versions.order(:id).first
```

Comment:
This doesn't improve much I think. The reason why it looks like it can be potentially a long array is because `ProxyConfig.current_versions` is generic enough to be used alone, but combined with the other scopes, effectively, it's already returning only one record here.

Reply:
Maybe `.limit(1)` instead of `.first` could really help, because that would prevent from an unnecessary longer table scan.

Reply:
Never mind. AR will take care of it and virtually it will be same.

Reply:
The `order(:id)` could be removed, BTW. It won't change anything, though it won't do any harm either.

Reply:
> This doesn't improve much I think. The reason why it looks like it can be potentially a long array is because ProxyConfig.current_versions is generic enough to be used alone, but combined with the other scopes, effectively, it's already returning only one record here.

I know. This is what https://github.com/3scale/porta/pull/2230 is for 😉 But it is a huge PR and I wanted to extract small cleanups in here first 😄 

> Maybe .limit(1) instead of .first could really help, because that would prevent from an unnecessary longer table scan.
> Never mind. AR will take care of it and virtually it will be same.

You replied yourself 😄 💪🏼 

> The order(:id) could be removed, BTW. It won't change anything, though it won't do any harm either.

I do not think so. When I was doing https://github.com/3scale/porta/pull/2256#discussion_r525084068 I could see that there can be more than 1 result and that the order in Oracle and Postgres matters 😄

Reply:
Btw, it is not only 1 record here, it is one record per version, proxy_id and version, and here we are not filtering by proxy_id but basically all of them

Reply:
my bad, it filters by proxy 😉